src/typescript-worker/completions.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import Fuse from 'fuse.js';
+import type {
+  GetCompletionsAtPositionOptions,
+  SymbolDisplayPart,
+} from 'typescript';
+import {
+  EditorCompletion,
+  EditorCompletionDetails,
+  EditorCompletionMatch,
+  EditorToken,
+  WorkerConfig,
+} from '../shared/worker-api.js';
+import {getWorkerContext} from './worker-context.js';
+
+/**
+ * Query completions from the Language Service, and sort them by
+ * relevance for user to use.
+ * */
+export const queryCompletions = async (
+  filename: string,
+  fileContent: string,
+  tokenUnderCursor: EditorToken,
+  cursorIndex: number,
+  config: WorkerConfig
+): Promise<EditorCompletion[]> => {
+  const workerContext = getWorkerContext(config);
+
+  const languageService = workerContext.languageServiceContext.service;
+  const languageServiceHost = workerContext.languageServiceContext.serviceHost;
+  const searchWordIsPeriod = tokenUnderCursor.string === '.';
+
+  const options = {} as GetCompletionsAtPositionOptions;
+  if (searchWordIsPeriod) {
+    options.triggerCharacter = '.';
+  }
+
+  // Update language service status so that the file is up to date
```

Comment:
Yeah, that's exactly the reason. If we are relying on the build step and it's state, we get rubbish completion, since it might lose the context for the completion. This way we can ensure that we have the actual current file context to give TS LS